kong/plugins/session/storage/kong.lua

```diff
@@ -10,103 +28,112 @@ local storage = {}
 storage.__index = storage
 
 
-function storage.new(session)
-  return setmetatable({
-    session     = session,
-    encode      = session.encoder.encode,
-    decode      = session.encoder.decode,
-  }, storage)
+local function load_session_from_db(key)
+  return kong.db.sessions:select_by_session_id(key)
 end
 
 
-local function load_session(id)
-  return kong.db.sessions:select_by_session_id(id)
+local function load_session_from_cache(key)
```

Comment:
just a thought, since this falls back to loading from db in case of cache miss, should we call it load_session_**with**_cache?

Reply:
I leace this as is, it kinda loads it from cache, if it is not found it loads it from db and puts it in cache, and then gets value from cache and returns it.